casper/src/main/scala/coop/rchain/casper/util/package.scala

```diff
@@ -0,0 +1,38 @@
+package coop.rchain.casper
+
+import com.google.protobuf.ByteString
+import coop.rchain.casper.protocol.{BlockMessage, Bond, RChainState}
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+package object util {
+  /*
+   * c is in the blockchain of b iff c == b or c is in the blockchain of the main parent of b
+   *
+   * This implies that the safety oracle must be run from the latest blocks towards the genesis
+   * for determining finality or else you might have cases in which blocks that are not in the main DAG
+   * will be not finalized.
+   */
+  // TODO: Move into BlockDAG and remove corresponding param once that is moved over from simulator
+  @tailrec
+  def isInMainChain(blocks: mutable.Map[ByteString, BlockMessage],
```

Comment:
Why specifically only take a mutable `Map`? If you drop the `mutable.` then either mutable or immutable `Map`s could be used.